token/program/src/processor.rs

```diff
@@ -439,6 +451,7 @@ impl Processor {
         }
 
         source_account.amount -= amount;
+        mint.supply -= amount;
```

Comment:
In a follow-up PR, can you please replace all the `-=` with `checked_sub()`.  Let's just use checked math everywhere in the program, even when it's obviously not necessary